src/model.jl

```diff
@@ -160,15 +207,26 @@ function _make_sets(
     demand_flexibility_enabled =
         isa(demand_flexibility, DemandFlexibility) && demand_flexibility.enabled
     if demand_flexibility_enabled
+        # if DOE profile is used, all buses with valid EIA ID are flexible
+        if demand_flexibility.enable_doe_flexibility
+            doe_flexible_bus_idx = sort(intersect(load_bus_idx, findall(case.bus_eiaid .> 0)))
+        end
```

Comment:
I think you need to handle the case where demand flexibility is enabled but DOE demand flexibility is not. I think it would probably be something like:

```
if demand_flexibility.enable_doe_flexibility
    doe_flexible_bus_idx = sort(intersect(load_bus_idx, findall(case.bus_eiaid .> 0)))
else
    doe_flexible_bus_idx = nothing
end
```